Lib/site.py

```diff
@@ -168,6 +169,14 @@ def addpackage(sitedir, name, known_paths):
     else:
         reset = False
     fullname = os.path.join(sitedir, name)
+    try:
+        st = os.lstat(fullname)
+    except OSError:
+        return
+    if ((getattr(st, 'st_flags', 0) & stat.UF_HIDDEN) or
```

Comment:
Which platforms support `UF_HIDDEN`? 

On macOS the flag is a hint that's ignored by command-line tools. 

E.g.:

```sh
ronald@Pelargir[0]% touch foo.txt                                                                   
[~/Projects/Forks/cpython/build/X]
ronald@Pelargir[0]% chflags hidden foo.txt                                                          
[~/Projects/Forks/cpython/build/X]
ronald@Pelargir[0]% ls -l                                                                           (gh-53502-long-times)cpython
total 0
-rw-r--r--  1 ronald  staff  0 Jan  2 21:10 foo.txt
ronald@Pelargir[0]% ls -lO                                                                          
total 0
-rw-r--r--  1 ronald  staff  hidden 0 Jan  2 21:10 foo.txt
```

This matches the documentation in `sys/stat.h`:

```c
#define UF_HIDDEN       0x00008000      /* hint that this item should not be */
                                        /* displayed in a GUI */
```